### PR TITLE
[nomerge] Nix __resource_path__

### PR DESCRIPTION
**Pull request by astronouth7303**
This just removes `__resource_path__` and makes all resources resolution relative to the Python module "virtual filesystem".

Note that this requires an `__init__.py` to exist next to any resource files.

### CODE REVIEW

File: ppb/vfs.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+"""
+Handles opening files from the Python "VFS".
+"""
+import logging
+from pathlib import Path
+import sys
+try:
+    import importlib.resources as impres
+except ImportError:
+    # Backport
```
Comment (nbraud) [inline ppb/vfs.py]: Please specify in a comment which Python versions require that.  (Presumably, only 3.6?)

File: ppb/vfs.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def _main_path():
+    main = sys.modules['__main__']
+    mainpath = getattr(main, '__file__')
+    if mainpath:
+        mainpath = Path(mainpath)
+        return mainpath.resolve().parent
+    else:
+        return Path.cwd()
```
Comment (nbraud) [inline ppb/vfs.py]: Document when `Path.cwd()` is expected to be used, and why it's a good choice.

File: ppb/vfs.py (hunk 1 above)
Comment (nbraud) [inline ppb/vfs.py]: Please specify in a comment which Python versions require that.  (Presumably, only 3.6?)

Comment (astronouth7303) [reply ppb/vfs.py]: Yeah, it's just 3.6

File: ppb/vfs.py (hunk 2 above)
Comment (astronouth7303) [reply ppb/vfs.py]: This is loosely based on the deleted `__resource_path__` implementation, where under a REPL `__main__` will refer to the console namespace and not have a `__file__`. (It used [`getfile`](https://github.com/python/cpython/blob/3.7/Lib/inspect.py) though)

**Comment by astronouth7303** [comment]
Note that I never really intended this to be merged, it's meant to be a demonstrative spike of "How would we do VFS?". I would much rather see the general resource revamp we've been discussing that incorporated this.

**Comment by astronouth7303** [comment]
Closing in favor of #306

**Pull request closed without merging**